common/nodes/database-schema.xml

```diff
@@ -405,6 +405,50 @@ CREATE TABLE ib_memberships (
   INDEX (switch, part_name, interface)
 );
 
+<!-- Firmware -->
+<!-- need to drop in reverse order of creation because of FK constraints -->
+DROP TABLE IF EXISTS firmware;
+DROP TABLE IF EXISTS firmware_model;
+DROP TABLE IF EXISTS firmware_make;
+DROP TABLE IF EXISTS firmware_imp;
+
+CREATE TABLE firmware_imp (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	name VARCHAR(255) NOT NULL,
+	INDEX (name),
+	CONSTRAINT unique_name UNIQUE (name)
```

Comment:
Don't need to name your constraints, you should be able to just say:
```
UNIQUE (name)
```

You might also be able to specify UNIQUE when you specify the column, like so:
```
name VARCHAR(255) NOT NULL UNIQUE,
```
But I would check that, since MySQL likes to ignore ANSI SQL syntax.